tests/foreman/cli/test_contentview.py

```diff
@@ -1012,6 +1013,89 @@ def test_positive_add_custom_repo_by_name(self):
+
+        @Assert: Module was added and has latest version.
+
+        @CaseLevel: Integration
+        """
+        module = {'name': 'versioned', 'version': '3.3.3'}
+        puppet_repository = make_repository({
+            u'content-type': u'puppet',
+            u'product-id': self.product['id'],
+            u'url': CUSTOM_PUPPET_REPO,
+        })
+        puppet_module = PuppetModule.list({
+            'search': 'name={0} and version={1}'.format(
+                module['name'], module['version'])
+        })[0]
+        Repository.synchronize({'id': puppet_repository['id']})
+        content_view = make_content_view({u'organization-id': self.org['id']})
+        ContentView.puppet_module_add({
+            u'content-view-id': content_view['id'],
+            u'name': puppet_module['name'],
+            u'author': puppet_module['author'],
+        })
+        cv_module = ContentView.puppet_module_list(
+            {'content-view-id': content_view['id']})
+        self.assertGreater(len(cv_module), 0)
+        self.assertIn(puppet_module['version'], cv_module[0]['version'])
+        self.assertIn('Latest', cv_module[0]['version'])
+
+    @tier2
+    @run_only_on('sat')
+    def test_positive_add_puppet_module_older_version(self):
+        """Add older version of puppet module to Content View by id/uuid
+
+        @id: 39654b3e-963f-4859-81f2-9992b60433c2
+
+        @Steps:
+
+        1. Upload/sync puppet repo with several versions of the same module
+        2. Add older version by id/uuid to CV
+
+        @Assert: Exact version (and not latest) was added.
+
+        @CaseLevel: Integration
+
+        @BZ: 1240491
+        """
+        module = {'name': 'versioned', 'version': '2.2.2'}
+        puppet_repository = make_repository({
+            u'content-type': u'puppet',
+            u'product-id': self.product['id'],
+            u'url': CUSTOM_PUPPET_REPO,
+        })
+        Repository.synchronize({'id': puppet_repository['id']})
+        puppet_module = PuppetModule.list({
+            'search': 'name={0} and version={1}'.format(
+                module['name'], module['version'])
+        })[0]
+        for by in ('uuid', 'id'):
+            with self.subTest(by=by):
+                content_view = make_content_view(
+                    {u'organization-id': self.org['id']})
+                ContentView.puppet_module_add({
+                    u'content-view-id': content_view['id'],
+                    by: puppet_module[by],
+                })
+                cv_module = ContentView.puppet_module_list(
+                    {'content-view-id': content_view['id']})
+                self.assertGreater(len(cv_module), 0)
+                self.assertEqual(cv_module[0]['version'], module['version'])
+                # Remove puppet module before next iteration
+                ContentView.puppet_module_remove({
+                    'content-view-id': content_view['id'],
+                    'uuid': puppet_module['uuid']
```

Comment:
So I've checked API, indeed puppet modules associated with CV have its own ids. But I have no idea where I am supposed to fetch them using hammer.
```
GET https://sat6.com/katello/api/content_views/66/content_view_puppet_modules/
{
  "total": 2,
  "subtotal": 2,
  "page": 1,
  "per_page": 20,
  "error": null,
  "search": null,
  "sort": {
    "by": null,
    "order": null
  },
  "results": [
...
    {
      "id": 28,
      "uuid": "daa047a1-3e3a-425c-bb09-659474619ac3",
      "name": "ui_test_variables",
      "author": "robottelo",
      "created_at": "2017-02-24 15:17:51 +0200",
      "updated_at": "2017-02-24 15:17:51 +0200",
      "computed_version": "0.2.0"
    }
  ]
}
```